src/clients/worker.rs

```diff
@@ -291,59 +307,71 @@ impl<C: golem_cloud_client::api::WorkerClient + Sync + Send> WorkerClient for Wo
             }
         });
 
-        let read_res = read.for_each(|message| async {
-            let message: Message = message.unwrap();
-
-            let msg = match message {
-                Message::Text(str) => {
-                    let parsed: serde_json::Result<InstanceConnectMessage> =
-                        serde_json::from_str(&str);
-                    Some(parsed.unwrap()) // TODO: error handling
-                }
-                Message::Binary(data) => {
-                    let parsed: serde_json::Result<InstanceConnectMessage> =
-                        serde_json::from_slice(&data);
-                    Some(parsed.unwrap()) // TODO: error handling
-                }
-                Message::Ping(_) => {
-                    debug!("Ignore ping");
-                    None
+        let read_res = read.for_each(|message_or_error| async {
+            match message_or_error {
+                Err(error) => {
+                    print!("Error reading message: {}", error);
                 }
-                Message::Pong(_) => {
-                    debug!("Ignore pong");
-                    None
-                }
-                Message::Close(_) => {
-                    info!("Ignore unexpected close");
-                    None
-                }
-                Message::Frame(_) => {
-                    info!("Ignore unexpected frame");
-                    None
-                }
-            };
-
-            match msg {
-                None => {}
-                Some(msg) => match msg.event {
-                    WorkerEvent::Stdout(StdOutLog { message }) => {
-                        print!("{message}")
+                Ok(message) => {
+                    let instance_connect_msg = match message {
+                        Message::Text(str) => {
+                            let parsed: serde_json::Result<InstanceConnectMessage> =
+                                serde_json::from_str(&str);
+                            Some(parsed.unwrap()) // TODO: error handling
+                        }
+                        Message::Binary(data) => {
+                            let parsed: serde_json::Result<InstanceConnectMessage> =
+                                serde_json::from_slice(&data);
+                            Some(parsed.unwrap()) // TODO: error handling
+                        }
+                        Message::Ping(_) => {
+                            debug!("Ignore ping");
+                            None
+                        }
+                        Message::Pong(_) => {
+                            debug!("Ignore pong");
+                            None
+                        }
+                        Message::Close(details) => {
+                            match details {
+                                Some(closed_frame) => {
+                                    print!("Connection Closed: {}", closed_frame);
```

Comment:
`info!`

Reply:
or `debug!`

Reply:
Hmm, print! is used in some other places to return back to users.

Reply:
It is used to output the actual stdout/stderr messages. Which is the goal.

Reply:
A close message is optional, and it is not a workerEvent:StdOut or WorkerEvent::StdErr. It is a mere String, and the type is not decided by us